core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/interceptor/AsyncRequestBodyHttpChecksumTrailerInterceptor.java

```diff
@@ -106,4 +112,9 @@ private static SdkHttpRequest updateHeadersForTrailerChecksum(Context.ModifyHttp
                         .putHeader(Header.CONTENT_LENGTH,
                                    Long.toString(chunkLength + checksumContentLength)));
     }
+
+    private static boolean isFileAsyncRequestBody(Context.ModifyHttpRequest context) {
+        return context.asyncRequestBody().isPresent() && context.asyncRequestBody().get() instanceof FileAsyncRequestBody;
+    }
```

Comment:
Hmm not a fan of this. Can we not fix this in a more generic way? What if a customer provides their own implementation that behaves exactly like FileAsyncRequestBody?

Reply:
Agree to this , will discuss to you offline.
We need to figure out the DEFAULT_CHUNK_SIZE which is dependent on AsyncRequestBody implementation

Reply:
Done